api/app/weather_models/fetch/predictions.py

```diff
@@ -202,6 +203,23 @@ async def fetch_model_predictions(model: ModelEnum, projection: ProjectionEnum,
     return await _fetch_model_predictions_by_station_codes(model, projection, station_codes)
 
 
+async def _fetch_delta_precip_for_prev_model_run(
+        model: ModelEnum,
+        station_codes: List[int],
+        prediction: WeatherModelPredictionValues,
+        prediction_model_run_timestamp):
+    """ Fetch the delta_precip for the station's prediction from the previous model run.
+    To be used in occasions where prediction_timestamp is not midnight but the prediction model run
+    is at hour 000 (and hence apcp is 0, so we must look at the delta_precip from the previous model run
+    in order to get a precipitation value)
+    """
+    # send the query
+    session = app.db.database.get_read_session()
+    prev_prediction = app.db.crud.weather_models.get_station_model_prediction_from_previous_model_run(
+        session, station_codes, model, prediction.prediction_timestamp, prediction_model_run_timestamp)
+    return prev_prediction.WeatherStationModelPrediction.delta_precip
```

Comment:
You have to handle null here - it's possible (at least with my database, that prev_prediction is None.